cmd/cluster/transferowner.go

```diff
@@ -267,7 +303,34 @@ func rolloutTelemeterClientPods(clientset *kubernetes.Clientset, namespace, sele
 	return nil
 }
 
-func verifyClusterPullSecret(clientset *kubernetes.Clientset, expectedPullSecret string) error {
+func comparePullSecretAuths(pullSecret *corev1.Secret, expectedAuths map[string]*amv1.AccessTokenAuth) error {
+	var errString strings.Builder
+	blue.Println("\nComparing pull-secret to expected auth sections...")
+	for akey, auth := range expectedAuths {
+		// Find the matching auth entry for this registry name in the cluster pull_secret data...
+		psTokenAuth, err := getPullSecretTokenAuth(akey, pullSecret)
+		if err != nil {
+			errString.WriteString(fmt.Sprintf("Failed to fetch expected auth['%s'] from cluster pull-secret, err:'%s'.\n", akey, err))
+		}
+		if auth.Auth() != psTokenAuth.Auth() {
+			errString.WriteString(fmt.Sprintf("Expected auth['%s'] does not match authToken found in cluster pull-secret.\n", akey))
+		} else {
+			green.Printf("Auth '%s' - tokens match\n", akey)
+		}
+		if auth.Email() != psTokenAuth.Email() {
+			errString.WriteString(fmt.Sprintf("Expected auth['%s'] does not match email found in cluster pull-secret.\n", akey))
+		} else {
+			green.Printf("Auth '%s' - emails match\n", akey)
+		}
+	}
+	if errString.Len() <= 0 {
+		return nil
+	} else {
+		return fmt.Errorf("%s", errString.String())
+	}
```

Comment:
Could we simplify this by using [errors.Join](https://pkg.go.dev/errors#Join)?

Reply:
Thanks!  Updated with join